### PR TITLE
DAOS-4240 docker: move to /opt/daos

### PR DESCRIPTION
**Pull request by johannlombardi**
Change prefix from /usr/local to /opt/daos to fix
build issue with rpath not being set.
Update documentation to skip submodule init/update.

Signed-off-by: Johann Lombardi <johann.lombardi@intel.com>

### CODE REVIEW

**Review by cdurf1 — COMMENTED** [review]
Approved with the incorporation of Johann's typo fix.

File: utils/docker/Dockerfile.centos.7 [hunk 1]
```diff
@@ -94,12 +94,10 @@ RUN useradd -u $UID -ms /bin/bash $USER
...
-RUN mkdir /mnt/daos
-RUN chown daos.daos /mnt/daos
-RUN mkdir /var/run/daos_server
-RUN chown daos.daos /var/run/daos_server
-RUN mkdir /var/run/daos_agent
-RUN chown daos.daos /var/run/daos_agent
+RUN mkdir -p /opt/daos && chown -R daos.daos /opt/daos
+RUN mkdir /mnt/daos && chown daos.daos /mnt/daos
+RUN mkdir /var/run/daos_server && chown daos.daos /var/run/daos_server
+RUN mkdir /var/run/daos_agent && chown daos.daos /var/run/daos_agent
```
Comment (brianjmurrell) [inline utils/docker/Dockerfile.centos.7:100]: Something to think about in the future:
```suggestion
RUN for dir in /opt/daos /mnt/daos /var/run/daos_server /var/run/daos_agent; do \
        mkdir $dir && chown daos.daos $dir; \
    done
```

**Review by brianjmurrell — APPROVED** [review]

File: utils/docker/Dockerfile.centos.7 [hunk 2]
```diff
@@ -195,16 +193,15 @@ WORKDIR /home/$USER
...
 RUN if [ "x$NOBUILD" = "x" ] ; then \
-    git submodule init && git submodule update; fi
-RUN if [ "x$NOBUILD" = "x" ] ; then \
-    scons --build-deps=yes install PREFIX=/usr/local; fi
+    scons --build-deps=yes install PREFIX=/opt/daos; fi
 
 # Set environment variables
-ENV LD_LIBRARY_PATH=/usr/local/lib:/usr/local/lib64:$LD_LIBRARY_PATH
-ENV PATH=/usr/local/bin:$PATH
+ENV LD_LIBRARY_PATH=/opt/daos/lib:/opt/daos/lib64:$LD_LIBRARY_PATH
```
Comment (jolivier23) [inline utils/docker/Dockerfile.centos.7:205]: this should not be necessary but probably doesn't hurt anything

File: utils/docker/Dockerfile.leap.15 [hunk 3]
```diff
@@ -175,16 +177,15 @@ WORKDIR /home/$USER
...
 RUN if [ "x$NOBUILD" = "x" ] ; then \
-    git submodule init && git submodule update; fi
-RUN if [ "x$NOBUILD" = "x" ] ; then \
-    scons --build-deps=yes install PREFIX=/usr/local; fi
+    scons --build-deps=yes install PREFIX=/opt/daos; fi
 
 # Set environment variables
-ENV LD_LIBRARY_PATH=/usr/local/lib:/usr/local/lib64:$LD_LIBRARY_PATH
-ENV PATH=/usr/local/bin:$PATH
+ENV LD_LIBRARY_PATH=/opt/daos/lib:/opt/daos/lib64:$LD_LIBRARY_PATH
```
Comment (jolivier23) [inline utils/docker/Dockerfile.leap.15:189]: Same comment, rpath should take care of this

File: utils/docker/Dockerfile.ubuntu.20.04 [hunk 4]
```diff
@@ -83,16 +81,14 @@ ARG NOBUILD
...
 RUN if [ "x$NOBUILD" = "x" ] ; then \
-  git submodule init && git submodule update; fi
-RUN if [ "x$NOBUILD" = "x" ] ; then \
-  scons --build-deps=yes install PREFIX=/usr/local; fi
+  scons --build-deps=yes install PREFIX=/opt/daos; fi
 
 # Set environment variables
-ENV LD_LIBRARY_PATH=/usr/local/lib:/usr/local/lib64:$LD_LIBRARY_PATH
-ENV PATH=/usr/local/bin:$PATH
+ENV LD_LIBRARY_PATH=/opt/daos/lib:/opt/daos/lib64:$LD_LIBRARY_PATH
```
Comment (jolivier23) [inline utils/docker/Dockerfile.ubuntu.20.04:92]: and again

**Review by jolivier23 — APPROVED** [review]

**Review by cdurf1 — APPROVED** [review]

**Pull request merged**